ckanext/security/authenticator.py

```diff
@@ -101,19 +101,19 @@ def authenticate(self, environ, identity):
         if throttle.is_locked():
             return None
 
-        if auth_user_name is None:
+        if ckan_auth_result is None:
             # Increment the throttle counter if the login failed.
             throttle.increment()
 
         # if the CKAN authenticator has successfully authenticated
         # the request and the user wasn't locked out above,
         # then check the TOTP parameter to see if it is valid
-        if auth_user_name is not None:
-            totp_success = self.authenticate_totp(environ, auth_user_name)
+        if ckan_auth_result is not None:
+            totp_success = self.authenticate_totp(environ, user_name)
             # if TOTP was successful -- reset the log in throttle
             if totp_success:
                 throttle.reset()
-                return totp_success
```

Comment:
@ThrawnCA this is also required since the `totp_success` was the username. So this change means that our override will always return the same value as `super(CKANLoginThrottle, self).authenticate` does.

Reply:
Ah, right, I overlooked that.